ray-operator/controllers/ray/common/service.go

```diff
@@ -118,6 +118,12 @@ func BuildServiceForHeadPod(ctx context.Context, cluster rayv1.RayCluster, label
 			Type:     defaultType,
 		},
 	}
+	if defaultType == "" || defaultType == corev1.ServiceTypeClusterIP {
+		// Use headless service for the Head Pod by default,
+		// since there is no sense for Head Pods to share a same virtual IP.
+		headService.Spec.ClusterIP = corev1.ClusterIPNone
```

Comment:
If a user has chosen to use the `NodePort` or `LoadBalancer` type, we don't make the service headless because they don't work with it.